synapse/handlers/federation.py

```diff
@@ -1666,14 +1670,47 @@ async def _resume_partial_state_room_sync(self) -> None:
 
         partial_state_rooms = await self.store.get_partial_state_room_resync_info()
         for room_id, resync_info in partial_state_rooms.items():
-            run_as_background_process(
-                desc="sync_partial_state_room",
-                func=self._sync_partial_state_room,
+            self._start_partial_state_room_sync(
                 initial_destination=resync_info.joined_via,
                 other_destinations=resync_info.servers_in_room,
                 room_id=room_id,
             )
 
+    def _start_partial_state_room_sync(
+        self,
+        initial_destination: Optional[str],
+        other_destinations: Collection[str],
```

Comment:
Could maybe use Olivier's `StrCollection` to avoid passing `other_destinations: str`. Not sure if that's worth the effort though :shrug:

Reply:
I'm going to leave the signature alone for now. It's supposed to match `_sync_partial_state_room` and I would rather not introduce more changes into a PR that's already proving difficult to follow.